app/workers/rapid_ready_for_decision/rrd_processor.rb

```diff
@@ -96,15 +97,33 @@ def icn
       account_record = account
       raise AccountNotFoundError, "for user_uuid: #{form526_submission.user_uuid} or their edipi" unless account_record
 
-      account_record.icn.presence
+      return account_record.icn.presence unless account_record.is_a?(Array)
+
+      # Multiple Account records should have the same ICN
+      icns = account_record.pluck(:icn).uniq.compact
+      return icns.first if icns.size == 1
+
+      raise MultipleIcnsFound, "for the user '#{form526_submission.user_uuid}': #{icns}"
```

Comment:
I might just not be understanding the problem with multiple ICNs and edipis or the consequences of using the wrong one (or when there are not consequences. I could imagine both cases, I could see it being better to go ahead with find_by instead of returning an error though.

What happens if the wrong account is used? Is there any possibility of it getting matched back to the correct thing through MPI or something?

Reply:
> What happens if the wrong account is used?

I'm not sure. Right now, we have no indication when this happens.
I'd be willing to send an alert notification to Slack instead of an error.

Reply:
It might be good to do that to start, and look at some of them.